src/Plugin/Condition/MediaSourceHasMimetype.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\islandora\Plugin\Condition;
+
+use Drupal\Core\Condition\ConditionPluginBase;
+use Drupal\Core\Form\FormStateInterface;
+use Drupal\file\Entity\File;
+
+/**
+ * Provides a 'Media Source' condition.
+ *
+ * @Condition(
+ *   id = "media_source_mimetype",
+ *   label = @Translation("Media Source Mimetype"),
+ *   context = {
+ *     "media" = @ContextDefinition("entity:media", required = FALSE, label = @Translation("Media"))
+ *   }
+ * )
+ */
+class MediaSourceHasMimetype extends ConditionPluginBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildConfigurationForm(array $form, FormStateInterface $form_state) {
+    $form['mimetype'] = [
+      '#title' => $this->t('Source Media Mimetype'),
+      '#type' => 'textfield',
+      '#default_value' => $this->configuration['mimetype'],
+    ];
+
+    return parent::buildConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    $this->configuration['mimetype'] = $form_state->getValue('mimetype');
+    parent::submitConfigurationForm($form, $form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function evaluate() {
+    foreach ($this->getContexts() as $context) {
+      if ($context->hasContextValue()) {
+        $entity = $context->getContextValue();
+        $mid = $entity->id();
+        if ($mid && !empty($this->configuration['mimetype'])) {
+          $source = $entity->getSource();
+          $source_file = File::load($source->getSourceFieldValue($entity));
+          if ($this->configuration['mimetype'] == $source_file->getMimeType()) {
+            return !$this->isNegated();
+          }
+        }
+      }
+    }
+    return $this->isNegated();
```

Comment:
After some digging I was able to find out why my content blocks stopped appearing, was a bit tricky as the cache had mislead me to think it was other problems. Checks for `isNegated` should not be used in `ConditionPlugins` as it causes a double negation, since the `ConditionManager` does the negation.
https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Condition%21ConditionManager.php/8.9.x
```php
public function execute(ExecutableInterface $condition) {
    if ($condition instanceof ConditionInterface) {
      $result = $condition
        ->evaluate();
      return $condition
        ->isNegated() ? !$result : $result;
    }
    throw new ExecutableException("This manager object can only execute condition plugins");
  }
```

Reply:
Ah, nice catch @nigelgbanks